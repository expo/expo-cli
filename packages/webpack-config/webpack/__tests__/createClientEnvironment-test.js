const { createEnvironmentConstants } = require('@expo/config');
const path = require('path');

const createClientEnvironment = require('../createClientEnvironment');
const getConfigAsync = require('../utils/getConfigAsync');
const getPathsAsync = require('../utils/getPathsAsync');
const normalizePaths = require('../utils/normalizePaths');

const projectRoot = path.resolve(__dirname, '../../tests/basic');
const mode = 'development';
const publicPath = '/';
const env = { projectRoot, mode };

describe('createClientEnvironment', () => {
  it('matches', async () => {
    const config = await getConfigAsync(env);
    const locations = await getPathsAsync(env);

    const publicAppManifest = createEnvironmentConstants(config, locations.production.manifest);
    const normalized = normalizePaths(publicAppManifest, value =>
      value.split('packages/webpack-config/').pop()
    );

    const environmentVariables = createClientEnvironment(env.mode, publicPath, normalized);

    // Add metro value
    expect(environmentVariables.__DEV__).toBe(true);
    // match trusted value
    expect(environmentVariables).toMatchSnapshot();
  });
});
