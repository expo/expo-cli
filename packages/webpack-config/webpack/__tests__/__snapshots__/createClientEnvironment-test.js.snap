// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createClientEnvironment matches 1`] = `
Object {
  "__DEV__": true,
  "process.env": Object {
    "APP_MANIFEST": "{\\"name\\":\\"basic\\",\\"slug\\":\\"basic\\",\\"privacy\\":\\"public\\",\\"sdkVersion\\":\\"33.0.0\\",\\"platforms\\":{\\"0\\":\\"ios\\",\\"1\\":\\"android\\",\\"2\\":\\"web\\"},\\"version\\":\\"1.0.0\\",\\"orientation\\":\\"portrait\\",\\"icon\\":\\"./assets/icon.png\\",\\"splash\\":{\\"image\\":\\"./assets/splash.png\\",\\"resizeMode\\":\\"contain\\",\\"backgroundColor\\":\\"#ffffff\\"},\\"updates\\":{\\"fallbackToCacheTimeout\\":0},\\"assetBundlePatterns\\":{\\"0\\":\\"**/*\\"},\\"description\\":\\"A Neat Expo App\\",\\"primaryColor\\":\\"#4630EB\\",\\"web\\":{\\"background_color\\":\\"#ffffff\\",\\"description\\":\\"A Neat Expo App\\",\\"dir\\":\\"auto\\",\\"display\\":\\"standalone\\",\\"lang\\":\\"en\\",\\"name\\":\\"basic\\",\\"orientation\\":\\"portrait\\",\\"prefer_related_applications\\":true,\\"related_applications\\":{},\\"short_name\\":\\"basic\\",\\"start_url\\":\\".\\",\\"theme_color\\":\\"#4630EB\\"}}",
    "NODE_ENV": "\\"development\\"",
    "PUBLIC_URL": "\\"/\\"",
  },
}
`;
